src/helpers/filterTypesHelper.ts

```diff
@@ -0,0 +1,168 @@
+  const { excludeDataProviderIds } = usePermanentFilter();
+
+  if (!filtersByQuery || filtersByQuery.config.active?.default === false) {
+    return [];
+  }
+
+  if (filtersByQuery.config.dateStart && filtersByQuery.config.dateEnd) {
+    delete filtersByQuery.config.dateEnd;
+  }
+
+  const filterTypes = Object.entries(filtersByQuery?.config ?? {}).map(([key, value]) => {
+    const filterType: FilterTypesProps = {
+      data: [],
+      label: getLabel('- Alle -'),
+      name: key as keyof FilterProps,
+      placeholder: getPlaceholder('- Alle -', query),
+      type: value.type
+    };
+
+    switch (key) {
+      case FILTER_KEYS.DATE_END:
+      case FILTER_KEYS.DATE_START:
+        filterType.label = getLabel(key);
+        filterType.data = [
+          {
+            name: QUERY_VARIABLES_ATTRIBUTES.DATE_START,
+            placeholder: getPlaceholder(FILTER_KEYS.DATE_START, query)
+          },
+          {
+            name: QUERY_VARIABLES_ATTRIBUTES.DATE_END,
+            placeholder: getPlaceholder(FILTER_KEYS.DATE_END, query)
+          }
+        ];
+        break;
+      case FILTER_KEYS.DATA_PROVIDER:
+        filterType.data = dropdownEntries(
+          query,
+          queryVariables,
+          data,
+          excludeDataProviderIds,
+          false
+        )?.map((entry: any) => ({
+          ...entry,
+          filterValue: entry.value
+        }));
+        filterType.label = getLabel(key);
+        filterType.name = QUERY_VARIABLES_ATTRIBUTES.DATA_PROVIDER;
+        filterType.placeholder = getPlaceholder(key, query);
+        break;
+      case FILTER_KEYS.CATEGORY:
+        filterType.data = dropdownEntries(
+          query,
+          queryVariables,
+          categories,
+          excludeDataProviderIds,
+          false
+        )?.map((entry: any) => ({
+          ...entry,
+          filterValue: entry.id
+        }));
+        filterType.label = getLabel(key);
+        filterType.name = QUERY_VARIABLES_ATTRIBUTES.CATEGORY_ID;
+        filterType.placeholder = getPlaceholder(key, query);
+        break;
+      case FILTER_KEYS.SAVEABLE:
+        filterType.label = getLabel(key);
+        filterType.placeholder = getPlaceholder(key, query);
+        filterType.checked = false;
+        break;
+      case FILTER_KEYS.RADIUS_SEARCH:
+        filterType.label = getLabel(key);
+        filterType.placeholder = getPlaceholder(key, query);
+        break;
+      case FILTER_KEYS.LOCATION:
+        filterType.data = dropdownEntries(
+          query,
+          queryVariables,
+          locations,
+          excludeDataProviderIds,
+          true
+        )?.map((entry: any) => ({
+          ...entry,
+          filterValue: entry.value
+        }));
+        filterType.label = getLabel(key);
+        filterType.name = QUERY_VARIABLES_ATTRIBUTES.LOCATION;
+        filterType.placeholder = getPlaceholder(key, query);
+        break;
+      default:
+        break;
```

Comment:
couldn't these two be default instead of having them in each case?

```
filterType.label = getLabel(key);
filterType.placeholder = getPlaceholder(key, query);
```

Reply:
Yes, that makes sense! 👍